### PR TITLE
Fix #6497 (opflow.expectations ignore coefficients)

### LINKED ISSUE #6497: Coefficients in Opflow Expectations are handled inconsistently
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?

In opflow, states, operators, and expectation value can have coefficients. For example the following expression has a coefficient of 2 for the state and a the measurement operator a coefficient of 3:
```python
exp = 3 * ~StateFn(Z) @ (2 * One)
```
which should evaluate to
```python
exp = <2 * One | 3 * Z | 2 * One> = 12 * <One | Z | One> = 12 * -1 = -12
```
Depending on which expectation value converter is used, opflow gives 3 different results for the above expression, see the code example below.

**Note:** This bug prevents the usage of `AerPauliExpectation` for gradients and the QFI and is related to Qiskit/qiskit-machine-learning#59.

### Steps to reproduce the problem

```python
from qiskit.providers.aer import QasmSimulator
from qiskit.opflow import *

exp = 3 * ~StateFn(Z) @ (2 * One)
result = exp.eval()
print('Plain eval:', result)  # -12 (correct)

result = PauliExpectation().convert(exp).eval()
print('PauliExpectation:', result)  # -4 (global factor 3 missing)

result = MatrixExpectation().convert(exp).eval()
print('MatrixExpectation:', result)  # -12 (correct)

sampler = CircuitSampler(QasmSimulator())
result = sampler.convert(AerPauliExpectation().convert(exp)).eval()
print('AerPauliExpectation:', result)  # -1 (both factors, 2 and 3, missing)
```

### What is the expected behavior?

All should give -12.

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Fix #6497

### CODE REVIEW

File: test/python/opflow/test_aer_pauli_expectation.py [hunk 1]
```diff
@@ -258,6 +255,13 @@ def test_pauli_expectation_non_hermite_op(self):
         exp = ~StateFn(1j * Z) @ One
         self.assertEqual(self.sampler.convert(self.expect.convert(exp)).eval(), 1j)
 
+    def test_expectation_with_coeff(self):
+        """Test AerPauliExpectation with coefficients."""
+        exp = 3 * ~StateFn(Z) @ (2 * One)
+        expected = exp.eval()
```
Comment (woodsp-ibm) [inline test/python/opflow/test_aer_pauli_expectation.py:261]: Just a minor comment, if the expected answer is easily known would it not be better to hard code it here? The assert as it stands, i.e. equal to exp.eval() could be in error and equally wrong - so things would be self consistent in opflow but not what we really expect.

File: test/python/opflow/test_pauli_expectation.py [hunk 2]
```diff
@@ -254,6 +254,13 @@ def test_list_pauli_sum_op(self):
         self.assertIsInstance(observable[0][0][0].primitive, PauliSumOp)
         self.assertIsInstance(observable[0][1][0].primitive, PauliSumOp)
 
+    def test_expectation_with_coeff(self):
+        """Test PauliExpectation with coefficients."""
+        exp = 3 * ~StateFn(Z) @ (2 * One)
+        expected = exp.eval()
```
Comment (woodsp-ibm) [inline test/python/opflow/test_pauli_expectation.py:260]: Similar comment on expected value.

File: test/python/opflow/test_aer_pauli_expectation.py (hunk 1 above)
Comment (ikkoham) [reply test/python/opflow/test_aer_pauli_expectation.py:261]: Thanks. If both are broken at the same time, this test won't detect it. I will change to hard code here.

**Review by woodsp-ibm — DISMISSED** [review]

**Review by woodsp-ibm — DISMISSED** [review]

**Review by Cryoris — APPROVED** [review]
Thanks a lot for this PR @ikkoham, this will also enable us to use the `AerPauliExpectation` to evaluate gradients and QFIs faster 👍🏻

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**